test/functional.js

```diff
@@ -943,23 +749,28 @@ describe('Functional tests using webpack', function() {
                 config.addEntry('page2', './preact/main');
 
                 // Move Vue.js code into its own chunk
-                config.addCacheGroup('vuejs', { test: /[\\/]node_modules[\\/]vue[\\/]/ });
+                const isVue2 = getVueVersion(config) === 2;
+                config.addCacheGroup('vuejs', {
+                    test: isVue2 ?
+                        /[\\/]node_modules[\\/]vue[\\/]/ :
+                        /[\\/]node_modules[\\/]@vue[\\/]/
+                });
```

Comment:
Quick explanation about why this is needed.
This is actually not directly related to Webpack 5 but because of how `vue` is now splitted into multiple packages under the `@vue` scope.

The imported module when you `import vue` does not do much:

```js
import { warn } from '@vue/runtime-dom';
export * from '@vue/runtime-dom';

// This entry exports the runtime only, and is built as
const compile = () => {
    if ((process.env.NODE_ENV !== 'production')) {
        warn(`Runtime compilation is not supported in this build of Vue.` +
            ( ` Configure your bundler to alias "vue" to "vue/dist/vue.esm-bundler.js".`
                ) /* should not happen */);
    }
};

export { compile };
```

With Webpack 4 the cache group seemed to worked because it created a shared file with the code of the `compile` function and the declaration of the multiple exports from `@vue/runtime-dom` (without their actual code... so not really useful).

With Webpack 5 I'm guessing that the `compile` function gets tree shaked and nothing matches the `node_modules/vue` test of the cache group anymore and no shared file is created.